nir/src/main/scala/scala/scalanative/nir/MemoryLayout.scala

```diff
@@ -0,0 +1,70 @@
+package scala.scalanative.nir
+
+import MemoryLayout.PositionedType
+
+object MemoryLayout {
+
+  sealed abstract class PositionedType {
+    def size: Long
+    def offset: Long
+  }
+  case class Tpe(size: Long, offset: Long, ty: Type) extends PositionedType
+  case class Padding(size: Long, offset: Long)       extends PositionedType
+
+  def apply(tys: Seq[Type]): MemoryLayout = {
+    val (size, potys) = helper(tys, 0)
+
+    MemoryLayout(size, potys.reverse)
+  }
+  private def helper(tys: Seq[Type],
```

Comment:
Sorry, it wasn't clear. `private` is fine, it's just that typically those kind of methods are called `impl`.